scripts/protect.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+
+userAgent="GitHub Autodeploy Bot/1.1.0 (${WIKI_UA_EMAIL})"
+
+declare -A loggedin
+declare -a allWikis
+declare -a regexErrors=()
+declare -a protectErrors=()
+
+regex="^\.?/?lua/wikis/([a-z0-9]+)/(.*)\.lua$"
+
+if [[ -n "$1" ]] then
+  filesToProtect=$1
+elif [[ -n ${WIKI_TO_PROTECT} ]]; then
+  filesToProtect=$(find lua/wikis -type f -name '*.lua')
+else
+  echo "Nothing to protect"
+  exit 0
+fi
+
+allLuaFiles=$(find lua -type f -name '*.lua')
+
+fetchAllWikis() {
+  allWikis=$(
+    curl \
+      -s \
+      -b "$ckf" \
+      -c "$ckf" \
+      -H "User-Agent: ${userAgent}" \
+      -H 'Accept-Encoding: gzip' \
+      -X GET "https://liquipedia.net/api.php?action=listwikis" \
+      | gunzip \
+      | jq '.allwikis | keys[]' -r
+  )
+  # Don't get rate limited
+  sleep 4
+}
+
+checkForLocalVersion() {
+  if [[ $2 == "commons" ]]; then
+    hasNoLocalVersion=false
+  elif [[ $allLuaFiles == *"lua/wikis/${2}/${1}.lua"* ]] || [[ $filesToProtect == *"lua/wikis/${2}/${1}.lua"* ]]; then
+    hasNoLocalVersion=false
+  else
+    hasNoLocalVersion=true
+  fi
+}
+
+protectPage() {
+  page="Module:${1}"
+  wiki=$2
+  protectOptions=$3
+  protectMode=$4
+  echo "...wiki = $wiki"
+  echo "...page = $page"
+  wikiApiUrl="${WIKI_BASE_URL}/${wiki}/api.php"
+  ckf="cookie_${wiki}.ck"
+
+  if [[ ${loggedin[${wiki}]} != 1 ]]; then
```

Comment:
a lot of this stuff is actually shared with the deploy script, maybe move it to a library file. Could have library functions to get tokens, to log in, to parse the comment, etc.

Reply:
will refactor and include #5706 too